### PR TITLE
feat(no-v-text-v-html-on-component): add `allow` option

### PR DESCRIPTION
**Pull request by waynzh**
Fixes https://github.com/vuejs/eslint-plugin-vue/issues/2321

This is my first time contributing to this project, so feel free to provide any feedback or suggestions～

### CODE REVIEW

File: docs/rules/no-v-text-v-html-on-component.md [hunk 1]
```diff
@@ -36,7 +36,41 @@ If you use v-text / v-html on a component, it will overwrite the component's con
...
+</eslint-code-block>
+
+<eslint-code-block :rules="{'vue/no-v-text-v-html-on-component': ['error', { allow: ['router-link', 'nuxt-link'] }]}">
+
+```vue
+<!-- ✗ BAD -->
+<template>
+  <MyComponent v-html="content"></MyComponent>
+</template>
+```
```
Comment (FloEdelmann) [inline docs/rules/no-v-text-v-html-on-component.md:71]: Let's merge these:

````suggestion
```vue
<template>
  <!-- ✓ GOOD -->
  <router-link v-html="content" />
  <NuxtLink v-html="content" />
  
  <!-- ✗ BAD -->
  <MyComponent v-html="content" />
</template>
```
````

File: lib/rules/no-v-text-v-html-on-component.js [hunk 2]
```diff
@@ -15,21 +15,50 @@ module.exports = {
...
+      allowedComponent = [...context.options[0].allow]
+    }
+
+    /**
+     * Check whether the given node is an allowed component or not.
+     * @param {VElement} node The start tag node to check.
+     * @returns {boolean} `true` if the node is an allowed component.
+     */
+    function isAllowedComponent(node) {
+      return allowedComponent.includes(node.rawName)
```
Comment (FloEdelmann) [inline lib/rules/no-v-text-v-html-on-component.js:52]: This should also allow `NuxtLink` if `nuxt-link` is in the `allow` list, and vice-versa. Please also add test cases for that and adjust the documentation.

See #2314 for a similar implementation.

Comment (waynzh) [reply lib/rules/no-v-text-v-html-on-component.js:52]: thanks for the reference, that makes sense!

**Review by FloEdelmann — APPROVED** [review]
Looks good to me :slightly_smiling_face: 
Thanks for your contribution!

**Review by ota-meshi — COMMENTED** [review]
LGTM! Thank you! Sorry for the late review.

**Pull request merged**